lxd/cluster/gateway.go

```diff
@@ -196,19 +197,37 @@ func (g *Gateway) HandlerFuncs(nodeRefreshTask func(*APIHeartbeat), trustedCerts
 			}
 
 			// Look for time skews
-			if heartbeatData.Time.Add(5 * time.Second).Before(time.Now().UTC()) {
+			now := time.Now().UTC()
+
+			if heartbeatData.Time.Add(5 * time.Second).Before(now) {
 				if !g.timeSkew {
-					logger.Warn("Time skew detected between leader and local", log.Ctx{"leaderTime": heartbeatData.Time, "localTime": time.Now().UTC()})
+					logger.Warn("Time skew detected between leader and local", log.Ctx{"leaderTime": heartbeatData.Time, "localTime": now})
+
+					err := g.Cluster.UpsertWarning("", "", -1, -1, db.WarningClusterTimeSkew, fmt.Sprintf("leaderTime: %s, localTime: %s", heartbeatData.Time, now))
+					if err != nil {
+						logger.Warn("Failed to create cluster time skew warning")
+					}
 				}
 				g.timeSkew = true
-			} else if heartbeatData.Time.Add(-5 * time.Second).After(time.Now().UTC()) {
+			} else if heartbeatData.Time.Add(-5 * time.Second).After(now) {
```

Comment:
@monstermunchkin could you merge this stanza condition with the one above as an OR statement to reduce the duplication here?